### ISSUE TITLE
MLIR converter quantizes conv2d before bias when dilaition rate > 1

### ISSUE THREAD

**Issue by reuvenperetz** [tensorflow/tensorflow#50747]
### 1. System information

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- TensorFlow installation (pip package or built from source): pip
- TensorFlow library (version, if pip package or github SHA, if built from source): 2.5.0
### 2. Code

Provide code to help us reproduce your issues using one of the following options:

#### Option A: Reference colab notebooks

Reference [TensorFlow Model Colab](https://colab.research.google.com/drive/1Pp0ZwbHTz3QjC2YxTn6l2kIro_DNxDqx?usp=sharing): Demonstrate how to build your TF model, convert to TFLite, and infer.

### 3. Failure after conversion
If the conversion is successful, but the generated model is wrong, then state what is wrong:
When creating a Conv2D with a dilation rate that is greater than 1, and the Conv2D has a bias, the MLIR converter splits the bias to a separate layer, and quantizes the tensor between them, thus causing accuracy degradation.

### 5. (optional) Any other info / logs
![image](https://user-images.githubusercontent.com/44209964/125436409-85a7a2e9-3158-4622-846e-baf1e08eae75.png)

**Comment by tilakrayal** [comment]
@reuvenperetz ,

We see that the issue [template](https://github.com/tensorflow/tensorflow/issues/new/choose) has not been filled, could you please do so as it helps us analyse the issue.Thanks!

**Comment by xhae** [comment]
Reproducing steps with the dedicated model would be really helpful!

**Comment by reuvenperetz** [comment]
### 1. System information

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- TensorFlow installation (pip package or built from source): pip
- TensorFlow library (version, if pip package or github SHA, if built from source): 2.5.0

### 2. Code

Provide code to help us reproduce your issues using one of the following options:

```python

import pathlib

import numpy as np
import tensorflow as tf

print(tf.__version__)

i = tf.keras.layers.Input(shape=(32, 32, 3))
o = tf.keras.layers.Conv2D(3, 3, dilation_rate=2)(i)
model = tf.keras.Model(inputs=[i], outputs=[o])

converter = tf.lite.TFLiteConverter.from_keras_model(model)

def representative_data_gen():
    yield [np.random.random((1, 32, 32, 3)).astype(np.float32)]


converter.representative_dataset = representative_data_gen
converter.optimizations = [tf.lite.Optimize.DEFAULT]
tflite_model = converter.convert()

tflite_models_dir = pathlib.Path("/tmp/")
tflite_model_file = tflite_models_dir / "dilation_model.tflite"
tflite_model_file.write_bytes(tflite_model)

interpreter = tf.lite.Interpreter(model_content=tflite_model)
interpreter.allocate_tensors()
input_details = interpreter.get_input_details()[0]

interpreter.set_tensor(input_details["index"], np.random.random((1, 32, 32, 3)).astype(np.float32))
interpreter.invoke()
output_details = interpreter.get_output_details()[0]
output = interpreter.get_tensor(output_details["index"])[0]

```



### 3. Failure after conversion
If the conversion is successful, but the generated model is wrong, then state what is wrong:

- Model produces wrong results and/or has lesser accuracy.

When creating a Conv2D with a dilation rate that is greater than 1, and the Conv2D has a bias, the MLIR converter splits the bias to a separate layer, and quantizes the tensor between them, thus causing accuracy degradation.

### 4. (optional) RNN conversion support
Not relaevant

### 5. (optional) Any other info / logs
![image](https://user-images.githubusercontent.com/44209964/126648418-0eda4143-9b85-44c7-9c21-1a1ba2bdef7a.png)

**Comment by pkgoogle** [comment]
Hi @abattery,

Can you please take a look at this? Thanks.

**Comment by pkgoogle** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The TFLite team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the TFLite space.
